tests/nn/data_parallel/test_fsdp_freezing_weights.py

```diff
@@ -118,21 +148,30 @@ def temp_files():
 
 
 @skip_if_single_gpu
-def test_freezing_weights(temp_files):
+@pytest.mark.parametrize("nested_trunk", ["nested_trunk", "simple_trunk"])
+def test_freezing_weights(temp_files, nested_trunk):
+    with_nested_trunk = nested_trunk == "nested_trunk"
+
     world_size = 2
     # DDP
-    fsdp = False
+    with_fsdp = False
     freezing_method = "requires_grad"
```

Comment:
nit and pet peeve of mine, but 
```
class FreezingMethod(str, Enum):
   ToNone = "to_none"
   RequiresGrad = "requires_grad"
```
removes the need for your asserts, adds type checking in different places, and makes it easy for a newcomer to know what are the supported methods without looking for asserts

Reply:
(it systematically prevents silent fall through for mismatches, even if you don't think of adding an assert everywhere, can be useful..)

Reply:
done. it is so much better! thanks for the suggestion!